bindings/c/test/apitester/TesterApiWorkload.cpp

```diff
@@ -126,11 +126,11 @@ fdb::Key ApiWorkload::randomNotExistingKey() {
 fdb::Key ApiWorkload::randomExistingKey() {
 	fdb::Key genKey = randomKeyName();
 	fdb::Key key = store.getKey(genKey, true, 1);
-	if (key != store.endKey()) {
+	if (key != store.endKey() && key.compare(0, keyPrefix.size(), keyPrefix) == 0) {
```

Comment:
I am afraid the probability of selecting a randomKey will be significantly reduced by this implementation. I would rather suggest using separate store instances for each tenant.